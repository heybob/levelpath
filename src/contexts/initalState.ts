import { Text } from "../shared/constants"

export const initialState = {
    isLoading: false, 
    data: [[]], 
    paginatedData: [], 
    pageNum: 1, 
    totalPages: 0,
    pageSize: 50, 
    today: new Date('December 17, 1995 03:24:00'),
    apiError: false,
    pageTitle: Text.headerTitle
}