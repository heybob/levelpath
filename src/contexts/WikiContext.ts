import React from "react";
import { createContext } from "react";
import { initialState } from "./initalState";
import { WikiState } from "../shared/types";

export const  WikiContext = createContext<{ 
    state: WikiState; dispatch: React.Dispatch<any>;}>(
        {state: initialState, dispatch: () => null});