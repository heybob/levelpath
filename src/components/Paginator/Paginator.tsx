import React from "react"
import { useContext, useEffect, useState } from "react"
import { WikiContext } from "../../contexts/WikiContext"
import './Paginator.css'
import { Button } from "../Button/Button"

type PaginatorProps = {
    handlePageChange: Function
}

export function Paginator({handlePageChange}: PaginatorProps) {
    const {state} = useContext(WikiContext);
    const {pageNum, totalPages}  = state;
    const [rows, setRows] = useState<React.JSX.Element[]>([]);

    useEffect(() => {
    }, [rows]);

    useEffect(() => {
        function generatePagination() {
            let newRows = [];
            if(totalPages > 0) {
                for(let i = 1; i <= totalPages; i++) {
                    newRows.push((<Button key={i} name={i.toString()} clickHandler={() => handlePageChange(i)} disabled={pageNum === i} />))
                }
                setRows(newRows);
            }
        }
        generatePagination();
    }, [totalPages, pageNum, handlePageChange])

    return (
        <div className='paginator'>
            {rows}
        </div>
    )
}