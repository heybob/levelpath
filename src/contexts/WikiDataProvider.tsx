import React from 'react'
import { useReducer} from "react";
import { wikiReducer } from "../components/App/reducer";
import { initialState } from "./initalState";
import { WikiContext } from "./WikiContext";

type Props = {
    children: React.ReactNode
  }

export function WikiDataProvider({children}: Props) {

    const [state, dispatch] = useReducer(wikiReducer, initialState);

    return (
        <WikiContext.Provider value={{state, dispatch}}>{children}</WikiContext.Provider>
    );

}