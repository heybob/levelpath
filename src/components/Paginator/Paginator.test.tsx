import renderer from 'react-test-renderer';
import { WikiDataProvider } from '../../contexts/WikiDataProvider';
import { Paginator } from "./Paginator"

describe('Testing Header Component', () => {
  it('It should render Header Title', () => {
        const handlePagechange = jest.fn();
      const tree = renderer.create(
      <WikiDataProvider>
            <Paginator handlePageChange={handlePagechange}/>
        </WikiDataProvider>
      ).toJSON()
    expect(tree).toMatchSnapshot();
  });
});
