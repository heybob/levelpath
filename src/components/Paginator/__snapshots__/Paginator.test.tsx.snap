// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component It should render Header Title 1`] = `
<div
  className="paginator"
/>
`;
